pkg/armadaevents/events_util.go

```diff
@@ -0,0 +1,139 @@
+package armadaevents
+
+import (
+	"encoding/json"
+	"errors"
+	time "time"
+)
+
+type RawES_Event struct {
+	Created *time.Time
+	Event   *json.RawMessage `json:"Event"`
+}
+
+func (ev *EventSequence_Event) UnmarshalJSON(data []byte) error {
```

Comment:
Is there no better way of unmarshalling this? I remember having similar issues unmarshalling oneof fields. I worked around the issue by using different unmarshallers; see: 
https://github.com/G-Research/armada/blob/master/internal/testsuite/app.go#L362

Did you try this? If there's no nice way of doing this, please at least add some tests.

Further, this function throws silently discards unmarshalling errors. For example, trying to unmarshall a malformed `EventSequence_Event_ReprioritiseJob` would result in `errors.New("could not determine EventSequence_Event.Event type for`. Is there a way to distinguish between these errors?

Reply:
I've redone this in commit 51557471 - it should now be faster and sturdier - please take a look. I did try several variations of using the unmarshallers as used in internal/testsuite/app.go, and they always quietly dropped the interface member. Also, it now delineates the unmarshaling error for specific struct types,, as opposed to the overall EventSequence_Event input being mangled, or if it cannot determine the underlying type.

The start of a unit test (it needs more of the many concrete structs underlying the EventSequence_Event.Event interface to be comprehensive) in in `pkg/armadaevents/events_util_test.go`.  I will add more - I've found some examples of constructing those big KubernetesObject objects, etc.